rasa/core/training/interactive.py

```diff
@@ -602,16 +630,38 @@ def _slot_history(tracker_dump: Dict[Text, Any]) -> List[Text]:
                 "value": a.get("action")}
                for a in predictions]
 
-    choices = ([{"name": "<create new action>", "value": OTHER_ACTION}] +
-               choices)
+    tracker = await retrieve_tracker(endpoint, sender_id)
+    evts = tracker.get("events", [])
+
+    session_actions_unique = []
+    session_actions_all = [a["name"] for a in _collect_actions(evts)]
+    [session_actions_unique.append(a) for a in session_actions_all if
+     a not in session_actions_unique]
+    old_actions = [action["value"] for action in choices]
+    new_actions = [{"name": action, "value": OTHER_ACTION + action} for action
+                   in session_actions_unique if action not in old_actions]
+
+    choices = ([{"name": "<create new action>", "value": NEW_ACTION}] +
+               new_actions + choices)
     question = questionary.select("What is the next action of the bot?",
                                   choices)
 
     action_name = await _ask_questions(question, sender_id, endpoint)
-    is_new_action = action_name == OTHER_ACTION
+    is_new_action = action_name == NEW_ACTION
 
     if is_new_action:
+        # create new action
         action_name = await _request_free_text_action(sender_id, endpoint)
+        if "utter_" in action_name:
+            utter_message = await _request_free_text_utterance(
+                sender_id, endpoint, action_name)
+            NEW_TEMPLATES[action_name] = {utter_message: "new message IL"}
```

Comment:
what's this "new message IL"?

Reply:
templates for some reason need to get the `utter_message` as dictionary key to save it correctly in the domain file. The "new message IL" is not used for anything, could also be "" or None. If the `utter_message` is passed as text and not dictionary it is saved letter by letter as multiple texts for the utter_template.

Reply:
```
utter_hi:
    - text: "h"
    - text: "i"
```

Reply:
hm we should be able to fix that somehow though, this format seems really weird

Reply:
This is how all text only templates from the domain file are converted to dictionaries in the `domain.py` file:
```            
for t in template_variations:
                # templates can either directly be strings or a dict with
                # options we will always create a dict out of them
                if isinstance(t, str):
                    validated_variations.append({"text": t})
```
I'll change it here to `NEW_TEMPLATES[action_name] = {utter_message: utter_message}` in order for it to be compatible with how templates are handled with other rasa code.